src/ripple/rpc/impl/ShardArchiveHandler.cpp

```diff
@@ -391,5 +581,42 @@ ShardArchiveHandler::remove(std::lock_guard<std::mutex>&)
     }
 }
 
+void
+ShardArchiveHandler::doRelease(std::lock_guard<std::mutex> const&)
+{
+    process_ = false;
+
+    timer_.cancel();
+    for (auto const& ar : archives_)
+        app_.getShardStore()->removePreShard(ar.first);
+    archives_.clear();
+
+    {
+        auto db = conn_->checkoutDb();
```

Comment:
No need for a LockedSociSession. conn_ is always used while holding the lock, getSession() can be used instead.

Reply:
Done